lua/highlite.lua

```diff
@@ -26,38 +26,62 @@ local PALETTE_HEX  = 1
 local NONE = "NONE"
 
 -- Get the color value of a color variable, or "NONE" as a default.
-local function get(color, index)
+local function get(color, index) -- {{{ †
 	if type(color) == 'table' and color[index] then
 		return color[index]
 	elseif type(color) == 'string' then
 		return color
 	else
 		return NONE
 	end
-end
+end --}}} ‡
+
+-- Add the 'blend' parameter to some highlight command, if there is one.
+local function blend(command, attributes) -- {{{ †
+	if attributes.blend then -- There is a value for the `highlight-blend` field.
+		command[#command + 1] = ' blend='..attributes.blend
+	end
+end --}}} ‡
 
 --[[ If using hex and 256-bit colors, then populate the gui* and cterm* args.
 	If using 16-bit colors, just populate the cterm* args. ]]
-local colorize = using_hex_or_256 and function(command, attributes) command[#command + 1] =
-	' ctermbg='..get(attributes.bg, PALETTE_256)
-	..' ctermfg='..get(attributes.fg, PALETTE_256)
-	..' guibg='..get(attributes.bg, PALETTE_HEX)
-	..' guifg='..get(attributes.fg, PALETTE_HEX)
-end or function(command, attributes) command[#command + 1] =
-	' ctermbg='..get(attributes.bg, PALETTE_ANSI)
-	..' ctermfg='..get(attributes.fg, PALETTE_ANSI)
-end
+local colorize = using_hex_or_256 and function(command, attributes) -- {{{ †
+	command[#command + 1] =
+		' ctermbg='..get(attributes.bg, PALETTE_256)
+		..' ctermfg='..get(attributes.fg, PALETTE_256)
+		..' guibg='..get(attributes.bg, PALETTE_HEX)
+		..' guifg='..get(attributes.fg, PALETTE_HEX)
+	blend(command, attributes)
+end or function(command, attributes)
+	command[#command + 1] =
+		' ctermbg='..get(attributes.bg, PALETTE_ANSI)
+		..' ctermfg='..get(attributes.fg, PALETTE_ANSI)
+	blend(command, attributes)
+end --}}} ‡
 
 -- This function appends `selected_attributes` to the end of `highlight_cmd`.
-local stylize = using_hex_or_256 and function(command, attributes)
-	command[#command + 1] = ' cterm='..attributes..' gui='..attributes
-end or function(command, attributes)
-	command[#command + 1] = ' cterm='..attributes
-end
+local stylize = using_hex_or_256 and function(command, style, color) -- {{{ †
+	command[#command + 1] = ' cterm='..style..' gui='..style
+
+	if color then -- There is an undercurl color.
+		command[#command + 1] = ' guisp='..get(color, PALETTE_HEX)
+	end
+end or function(command, style)
+	command[#command + 1] = ' cterm='..style
+end --}}} ‡
 
 -- Generate a `:highlight` command from a group and some attributes.
 local function highlight(highlight_group, attributes) -- {{{ †
+	-- The base highlight command
 	local highlight_cmd = {'hi! ', highlight_group}
+
+	-- Take care of special instructions for certain background colors.
+	if attributes[vim.o.background] then
```

Comment:
This is the feature implementation.

The rest is either logic simplification (for my sake as a maintainer), or efficiency improvements (to offset increased complexity of table key resolution).

Reply:
Additionally, if Neovim ever receives `background` options besides "light" and "dark", they will automatically be supported.